snapshot/snapshotfs/upload.go

```diff
@@ -663,7 +677,10 @@ func uploadDirInternal(
 		return "", fs.DirectorySummary{}, workItemErr
 	}
 
-	if err := u.processUploadWorkItems(workItems, dirManifest); err != nil && err != errCancelled {
+	errHandlingPolicy := policyTree.EffectivePolicy().ErrorHandlingPolicy
+	ignoreFileErrs := u.IgnoreReadErrors || (errHandlingPolicy.IgnoreFileErrors != nil && *errHandlingPolicy.IgnoreFileErrors)
```

Comment:
nit: for readability perhaps add a helper here

u.shouldIgnoreReadErrors(policyTree)

Reply:
good idea, will add